util/design/gen-flash-img.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Takes a compiled vmem image and processes it for flash.
+    Long term this should include both layers of ECC and scrambling,
+    The first version has only the truncated plaintext ECC.
+"""
+
+import re
+from pathlib import Path
+import argparse
+import secded_gen
+import math
+
+
+def add_plain_ecc(in_val: int) -> str:
+    result, m = secded_gen.ecc_encode("hamming", 64, in_val)
+
+    m_nibbles = math.ceil(m / 4)
+    result = format(result, '0' + str(16 + m_nibbles) + 'x')
+
+    # due to lack of storage space, the first nibble of the ECC is truncated
+    return result[1:]
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('infile')
+    parser.add_argument('outfile')
+    args = parser.parse_args()
+
+    # open original vmem and extract relevant content
+    try:
+        with Path(args.infile).open() as f:
+            vmem_orig = f.read()
```

Comment:
since this is a Pathlib obj, I think you could do `vmem_orig = Path(args.infile).read_text()` instead, since you are opening the file for reading  in text mode ([defaults of `open()`](https://docs.python.org/3/library/functions.html#open))

Reply:
ahhh, let me try that. Should I still bracket it with a `try / except` ?